### PR TITLE
Remove Bot API Related Deprecations

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
ToDo:

- [x] https://github.com/python-telegram-bot/python-telegram-bot/projects/9#card-88599701 - see #3584
- [x] https://github.com/python-telegram-bot/python-telegram-bot/projects/9#card-87927012
  - [x] https://github.com/python-telegram-bot/python-telegram-bot/pull/3530#discussion_r1096320653
  - [x] https://github.com/python-telegram-bot/python-telegram-bot/pull/3530#discussion_r1096320653
- [x] Adjust tests
- [x] merge #3853 into master

### CODE REVIEW

File: telegram/_message.py [hunk 1]
```diff
@@ -3624,7 +3604,7 @@ def text_markdown(self) -> str:
 
             * |custom_emoji_formatting_note|
 
-        .. deprecated:: 20.3
+        .. versionchanged:: NEXT.VERSION
             |custom_emoji_md1_deprecation|
```
Comment (harshil21) [inline telegram/_message.py:3608]: "deprecation" -> something else?

File: tests/test_official.py [hunk 2]
```diff
@@ -123,39 +116,7 @@ def ignored_param_requirements(object_name) -> Set[str]:
...
-        "thumb_height",
-        "thumb_width",
-    },
-    "InlineQueryResult(Article|Photo|Gif|Mpeg4Gif|Video|Contact|Document|Location|Venue)": {
-        "thumb_url",
-    },
-    "InlineQueryResult(Game|Gif|Mpeg4Gif)": {"thumb_mime_type"},
-    "answer_inline_query": {"switch_pm_text", "switch_pm_parameter"},
-}
+BACKWARDS_COMPAT_KWARGS = {}
```
Comment (harshil21) [inline tests/test_official.py:119]: clean 🔥

**Review by harshil21 — CHANGES_REQUESTED** [review]
damn that was a lot of code. Doc build is failing btw.

**Review by lemontree210 — COMMENTED** [review]
Amazing effort 👏🏻

File: telegram/_files/_basethumbedmedium.py [hunk 3]
```diff
@@ -48,10 +44,6 @@ class _BaseThumbedMedium(_BaseMedium):
             is supposed to be the same over time and for different bots.
             Can't be used to download or reuse the file.
         file_size (:obj:`int`, optional): File size.
-        thumb (:class:`telegram.PhotoSize`, optional): Thumbnail as defined by sender.
```
[the remark this answers is not in the document]
Comment (Bibo-Joshi) [reply telegram/_files/_basethumbedmedium.py:51]: TBH I didn't bother b/c it's a private class. I don't really know why it has a docstring in the first place :D

**Pull request merged**